cmd/relay/repomgr/repomgr.go

```diff
@@ -0,0 +1,398 @@
+package repomgr
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	atproto "github.com/bluesky-social/indigo/api/atproto"
+	"github.com/bluesky-social/indigo/atproto/identity"
+	atrepo "github.com/bluesky-social/indigo/atproto/repo"
+	"github.com/bluesky-social/indigo/atproto/syntax"
+	"github.com/bluesky-social/indigo/cmd/relay/events"
+	"github.com/bluesky-social/indigo/cmd/relay/models"
+	"github.com/ipfs/go-cid"
+	"go.opentelemetry.io/otel"
+)
+
+const defaultMaxRevFuture = time.Hour
+
+func NewRepoManager(directory identity.Directory, inductionTraceLog *slog.Logger) *RepoManager {
+	maxRevFuture := defaultMaxRevFuture // TODO: configurable
+	ErrRevTooFarFuture := fmt.Errorf("new rev is > %s in the future", maxRevFuture)
+
+	return &RepoManager{
+		userLocks:         make(map[models.Uid]*userLock),
+		log:               slog.Default().With("system", "repomgr"),
+		inductionTraceLog: inductionTraceLog,
+		directory:         directory,
+
+		maxRevFuture:           maxRevFuture,
+		ErrRevTooFarFuture:     ErrRevTooFarFuture,
+		AllowSignatureNotFound: true, // TODO: configurable
```

Comment:
should definitely not be `true` by default?

Reply:
I think we can maybe refine which versions of "not found" are allowed or not, but until then it has to broadly be on in prod and is at worst a warning.